### PR TITLE
Switch to H5_USE_110_API

### PR DESCRIPTION
**Pull request by avalentino**
Compatibility with HDF5 1.8.x is no longer needed, so we can switch to the newer API

### CODE REVIEW

**Review by FrancescAlted — APPROVED** [review]
Sounds good to me.

File: setup.py [hunk 1]
```diff
@@ -776,30 +776,8 @@ def find_runtime_path(self, locations=None):
     # Force the 1.8.x HDF5 API even if the library as been compiled to use the
```
Comment (ivilata) [inline setup.py:776]: LGTM, maybe this comment could be updated as well, and I think some ifdefs in `utils.h` may also be removed. Then there's `utilsextension.get_native_type`, but simplifying that may be more involved.

Thanks for the cleanup!

**Review by ivilata — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (avalentino) [reply setup.py]: Thanks @ivilata 
I have performed some cleanup.
I kindly ask you to give another look if you can.

**Review by ivilata — APPROVED** [review]
A very thorough cleanup indeed, LGTM!

**Pull request merged**